### ISSUE TITLE
News From Date attribute auto set to non empty value if special price is greater than 0

### ISSUE THREAD

**Issue by daniel-ifrim** [magento/magento2#5840]
### Preconditions

1. Develop branch or other Magento version. I'm on Magento 2.0.7
### Steps to reproduce

1. See https://github.com/magento/magento2/blob/develop/app/code/Magento/Catalog/Model/Attribute/Backend/Startdate.php
### Expected result

1. News From Date should not be related with Special Price
### Actual result

1. There should not be a link between news_from_date attribute and special price.
   I know special_from_date is related to special price but news from date is not.

Fix would be to change the backend of news from date from:
Magento\Catalog\Model\Attribute\Backend\Startdate
to seomthing else like:
Magento\Eav\Model\Entity\Attribute\Backend\Time\Created
or
Magento\Eav\Model\Entity\Attribute\Backend\Datetime

**Comment by oserediuk** [comment]
Hello! Could you provide steps to reproduce more clear?

**Comment by daniel-ifrim** [comment]
There is:

```
protected function _getValueForSave($object)
    {
        $attributeName = $this->getAttribute()->getName();
        $startDate = $object->getData($attributeName);
        if ($startDate === false) {
            return false;
        }
        if ($startDate == '' && $object->getSpecialPrice()) {
            $startDate = $this->_localeDate->date();
        }
        return $startDate;
    }

```

here:
https://github.com/magento/magento2/blob/develop/app/code/Magento/Catalog/Model/Attribute/Backend/Startdate.php

Attribute with `attribute_code` "news_from_date" has in 'backend' column in `eav_attribute` table
"Magento/Catalog/Model/Attribute/Backend/Startdate"

The snippet of code for news_to_date attribute will do:
- if the value of this attribute from product is empty
- and if $object->getSpecialPrice() is not empty
  than 'autocomplete' news_from_date with local value.
  This will happen only if the product has special price.

I know that "special_from_date" attribute uses this backend class.
But `special_from_date` attribute should have it's own backend class because it needs auto completion only when Special Price is not empty.
News from date attribute should not relate to special price.
If some merchant thinks News From Date 'auto completion' should depend on special price than they can code this condition in the collection that is used in New Products block or whatever other part of frontend code.

**Comment by cherreman** [comment]
Is there any update on this? In the meantime, we patched the Startdate class and added a check to see if the attribute is "special_from_date" before returning a new date.

**Comment by magento-engcom-team** [comment]
@daniel-ifrim, thank you for your report.
We were not able to reproduce this issue by following the steps you provided. If you'd like to update it, please reopen the issue.
We tested the issue on 2.1.9

**Comment by daniel-ifrim** [comment]
You don't have to reproduce it.
It's basic programming.

**Comment by PetraSwed** [comment]
Facing the same irritating issue. magento 2.1.8. Setting special price on a product (under Advanced Pricing) sets automatically "Set product as new From" to todays date.

**Comment by daniel-ifrim** [comment]
@PetraSwed They fixed it in another issues or in release 2.2.0
https://github.com/magento/magento2/issues/4387
Better search for the fix in 2.2.0
